test/Main.purs

```diff
@@ -179,87 +159,164 @@ main = do
+    ((rootDir </> dirFoo) `relativeTo` (rootDir </> dirBar))
+    "./../foo/"
+
+  test' "(rootDir </> dirBar) `relativeTo` (rootDir </> dirFoo) = ./../bar/"
+    ((rootDir </> dirBar) `relativeTo` (rootDir </> dirFoo))
+    "./../bar/"
+
+  test' "(rootDir </> dirBar) `relativeTo` (rootDir </> dirFoo </> dirFoo) = ./../../bar/"
+    ((rootDir </> dirBar) `relativeTo` (rootDir </> dirFoo </> dirFoo))
+    "./../../bar/"
+
+  test' "(rootDir </> dirBar) `relativeTo` (rootDir </> dirFoo </> dirFoo </> dirFoo) = ./../../../bar/"
+    ((rootDir </> dirBar) `relativeTo` (rootDir </> dirFoo </> dirFoo </> dirFoo))
+    "./../../../bar/"
+
+  test' "(rootDir </> dirBar </> dirBar) `relativeTo` (rootDir </> dirFoo) = ./../bar/bar/"
+    ((rootDir </> dirBar </> dirBar) `relativeTo` (rootDir </> dirFoo))
+    "./../bar/bar/"
 
-   (Just $ file $ reflectNonEmpty $ SProxy :: SProxy "image")
+  test' "(rootDir </> dirBar </> dirBar) `relativeTo` (rootDir </> dirFoo </> dirFoo) = ./../../bar/bar/"
+    ((rootDir </> dirBar </> dirBar) `relativeTo` (rootDir </> dirFoo </> dirFoo))
+    "./../../bar/bar/"
 
-  test' "sandbox - sandbox absolute dir to one level higher"
-    (unsafePartial $ fromJust $ sandbox (rootDir </> dirFoo) (rootDir </> dirFoo </> dirBar))
-    "./bar/"
+  test' "(rootDir </> dirBar </> dirFoo </> dirFoo) `relativeTo` (rootDir </> dirFoo </> dirFoo </> dirFoo) = ./../../../bar/foo/foo"
+    ((rootDir </> dirBar </> dirFoo </> dirFoo) `relativeTo` (rootDir </> dirFoo </> dirFoo </> dirFoo))
+    "./../../../bar/foo/foo/"
 
-  test "depth - negative"
-    (depth (parentDir $ parentDir $ parentDir $ currentDir)) (-3)
+  test' "(rootDir </> dirFoo </> dirBar </> dirBaz) `relativeTo` rootDir = ./foo/bar/baz/"
+    ((rootDir </> dirFoo </> dirBar </> dirBaz) `relativeTo` rootDir)
+    "./foo/bar/baz/"
+
+  test' "(rootDir </> dirFoo </> dirBar </> dirBaz) `relativeTo` (rootDir </> dirFoo) = ./bar/baz/"
+    ((rootDir </> dirFoo </> dirBar </> dirBaz) `relativeTo` (rootDir </> dirFoo))
+    "./bar/baz/"
+
+  test' "(rootDir </> dirFoo </> dirBar </> dirBaz) `relativeTo` (rootDir </> dirBaz) = ./../foo/bar/baz/"
+    ((rootDir </> dirFoo </> dirBar </> dirBaz) `relativeTo` (rootDir </> dirBaz))
+    "./../foo/bar/baz/"
+
+  test' "(rootDir </> dirBar </> dirFoo) `relativeTo` (rootDir </> dirBar) = ./foo/"
+    ((rootDir </> dirBar </> dirFoo) `relativeTo` (rootDir </> dirBar))
+    "./foo/"
+
+  test "rename - single level deep"
+    (rename (alterExtension (const Nothing)) (file (reflectNonEmpty $ SProxy :: SProxy "image.png")))
+    (file $ reflectNonEmpty $ SProxy :: SProxy "image")
+
+  test "sandbox - fail when relative path lies outside sandbox (above)"
+    (sandbox (rootDir </> dirBar) (parentOf currentDir))
+    Nothing
+
+  test "sandbox - fail when relative path lies outside sandbox (neigbouring)"
+    (sandbox (rootDir </> dirBar) (parentOf currentDir </> dirFoo))
```

Comment:
`(parentOf currentDir </> dirFoo)` is `(currentDir <..> dirFoo)` right?

Reply:
Yep, I think so!